django/http/response.py

```diff
@@ -153,13 +153,10 @@ def charset(self, value):
 
     def serialize_headers(self):
         """HTTP headers as a bytestring."""
-        def to_bytes(val, encoding):
-            return val if isinstance(val, bytes) else val.encode(encoding)
-
-        headers = [
-            (to_bytes(key, 'ascii') + b': ' + to_bytes(value, 'latin-1'))
+        headers = (
+            (key.encode("ascii") + b': ' + value.encode("latin-1"))
             for key, value in self.headers.items()
-        ]
+        )
```

Comment:
When using `str.join()` it is preferable to pass a `list` as it is slightly faster.

Reply:
You don't need to create a list, actually. You can just pass the generator expression through (no surrounding parentheses are needed).

Reply:
This has come up in the past when we were looking at removing unnecessary list comprehensions in favour of generators. IIRC, `str.join()` converts input to a list if it isn't already.

Reply:
For future reference, [here is the line](https://github.com/python/cpython/blob/aaa83cdfab6817446285e631232f64b394ac6791/Objects/unicodeobject.c#L10022) of interest.

And here are the docs for [`PySequence_Fast()`](https://docs.python.org/3/c-api/sequence.html#c.PySequence_Fast) which states:

> The `PySequence_Fast*` functions are thus named because they assume `o` is a `PyTupleObject` or a `PyListObject` and access the data fields of `o` directly.
> 
> As a CPython implementation detail, if `o` is already a sequence or list, it will be returned.

Reply:
Interesting, thanks!